### ISSUE TITLE
Provide a way to instantiate components with a method name, i.e. NewSomeComponent

### ISSUE THREAD

**Issue by bradleypeabody** [vugu/vugu#94]
It would be better practice to be able to instantiate components which have a New... method by calling that method. 

The main question is the syntax.  If we can come up with an appropriate syntax that isn't at odds with everything else and isn't too clumsy, this should be implemented.

**Comment by itstmyi** [comment]
Good  idea .
Now I have to add a “initialized bool” field in my Root structure to record the component's initialized status  . BeforeBuild method excutes before every rendering , so I have to keep the stauts and some values  by checking my  “initialized bool” field .

**Comment by tbe** [comment]
What about `<main:Component :args='argA, argB, argC'></main:Component>`?

As fields of an component struct have to be exported, it should be save to use lowercase `args`. This would solve the problem imho and has the major benefit, that this could be
- combined with normal fields behavior
- could eliminate reflection - if there is a `:args`, just assume  there is a `New(...)`

**Comment by bradleypeabody** [comment]
Yeah, interesting.  Possibly `<pkg:Comp vg-new="argA, argB, argC">`. (I've been generally trying to keep the convention of attributes with magic behavior starting with `vg-`.)

But other than the naming, there is an issue with the component lifecycle.  As it is right now, components can be re-used and they then have their properties set on them again.  The main purposes of this is not necessarily because making a struct instance takes time, but that if we keep the same struct instance and know that it has the same state we can cache the build output.  (And later, hopefully avoid re-syncing those DOM elements.). So if we did this I'm not sure what to do about the disparity of having NewX(A,B,C) the first time and then subsequent times keeping the same component and having to assign A,B,C as fields.  Needs more consideration to sort through this.

**Comment by tbe** [comment]
> . As it is right now, components can be re-used and they then have their properties set on them again.

The only thing i found is, that there is a static ID generated per component and the created instance (currently via `new`) is stored under this id and an iterkey. Did i miss something?

**Comment by bradleypeabody** [comment]
A component reference in the generated code looks like this:

```go
{
	vgcompKey := vugu.MakeCompKey(0x5E8BFE1B6CE0ADB9, vgiterkey)
	// ask BuildEnv for prior instance of this specific component
	vgcomp, _ := vgin.BuildEnv.CachedComponent(vgcompKey).(*Header)
	if vgcomp == nil {
		// create new one if needed
		vgcomp = new(Header)
		vgin.BuildEnv.WireComponent(vgcomp)
	}
	vgin.BuildEnv.UseComponent(vgcompKey, vgcomp)	// ensure we can use this in the cache next time around
	vgcomp.SomeField = someValue // field assignments here
	vgout.Components = append(vgout.Components, vgcomp)
	vgn = &vugu.VGNode{Component: vgcomp}
	vgparent.AppendChild(vgn)
}
```

The call to `CachedComponent` and the check for `vgcomp == nil` is what I was referring to above.  As you can see the field assignments are below the conditional `new()` call.   This proposal would essentially move the assignments up to be inside that `if` statement; so we'd need to work out what happens in the case where `vgcomp != nil` after calling `CachedComponent`
